src/app/modules/test-navigator/test-navigator.component.ts

```diff
@@ -169,27 +181,119 @@ export class TestNavigatorComponent implements OnInit, OnDestroy {
+    const contextNode = this.selectedNode;
+    const parentNode = contextNode.type === ElementType.Folder ? contextNode : contextNode.parent;
+    const payload: TreeViewerInputBoxConfig = {
+      indent: contextNode.type === ElementType.Folder,
+      validateName: (newName: string) => this.validateName(newName, type),
+      onConfirm: async (newName: string) => {
+        const newPath = contextNode.getDirectory() + newName;
+        const requestSuccessful = await this.sendCreateRequest(newPath, type);
+        if (requestSuccessful) {
+          /* const newNode = */ parentNode.addChild({children: [], name: newName, path: newPath, type: type });
+          parentNode.expanded = true;
+          // TODO programmatically select newNode
+          if (type === ElementType.File) {
+            this.messagingService.publish(NAVIGATION_OPEN, { name: newName, id: newPath });
+          }
+        }
+        return requestSuccessful;
+      },
+      iconCssClasses: type === ElementType.Folder ? 'fa-folder' : 'fa-file'
+    };
+    this.messagingService.publish(TREE_NODE_CREATE_AT_SELECTED, payload);
+  }
+
+  renameElement(): void {
+    const selectedNode = this.selectedNode;
+    const payload: InputBoxConfig = {
+      validateName: (newName: string) => this.selectedNode.dirty ?
+        { valid: false, message: 'cannot rename dirty files' } : this.validateName(newName, selectedNode.type),
+      onConfirm: async (newName: string) => {
+        const pathElements = selectedNode.id.split('/');
+        const newPath = pathElements.slice(0, pathElements.length - 1).join('/') + '/' + newName;
+        const requestSuccessful = await this.sendRenameRequest(newPath, selectedNode.id);
+        if (requestSuccessful) {
+          selectedNode.rename(newPath, newName);
+        }
+        return requestSuccessful;
+      }
+    };
+    this.messagingService.publish(TREE_NODE_RENAME_SELECTED, payload);
+  }
+
+  private validateName(newName: string, type: ElementType): { valid: boolean, message?: string } {
+    let result: { valid: boolean, message?: string } = { valid: true };
+    if (!this.pathValidator.isValid(newName)) {
+      result = { valid: false, message: this.pathValidator.getMessage(newName) };
+    }
+    if (!filterFor(this.filterState, {type: type, id: newName})) {
+      result = { valid: false, message: 'invalid (or currently filtered) file extension' };
+    }
+    return result;
+  }
+
+  private async sendCreateRequest(newPath: string, type: ElementType): Promise<boolean> {
+    let result = false;
+    try {
+      const response = await this.persistenceService.createResource(newPath, type);
+      let createdPath: string;
+      if (isConflict(response)) {
+        this.errorMessage = response.message;
+        createdPath = newPath;
+      } else {
+        createdPath = response;
+        result = true;
+      }
+      this.messagingService.publish(NAVIGATION_CREATED, { path: createdPath });
```

Comment:
I'd rather use something like WORKSPACE_ELEMENT_CREATED

Reply:
Sure, but note that these were pre-existing events which other modules rely on, i.e. we need to change it in those places later, as well.